inbox/optimisedS2S.xml

```diff
@@ -0,0 +1,123 @@
+  </p>
+<p> 
+The solution set out here is to eliminate all the intial handshaking and to start the S2S communication as if the handshaking had been correctly completed.  This cannot be used for communication between an arbitrary pair of servers, as in general the negotiation associated with the handshaking is vital for correctly determining a variety of parameters for use in the connection.    However, a pair of servers may operate by locally configuring information that would have been negotiated.   This enables the pair of servers to eliminate initial handshaking and data exchange.
+</p>
+</section1>
+<section1 topic='Requirements' anchor='reqs'>
+  <p>
+    This specification can be considered as a profile for server to server XMPP communication, to enable XMPP deployment over high latency links.   This profile MUST only be used where its use has been pre-agreed and configured for both participating servers.  
+  </p>
+</section1>
+<section1 topic='Glossary' anchor='glossary'>
+  <p>OPTIONAL.
+  </p>
+  <p> Sections without text to be removed prior to sending doc for XSF review.</p>
+</section1>
+<section1 topic='Use Cases' anchor='usecases'>
+  <p>STRONGLY RECOMMENDED.</p>
+</section1>
+<section1 topic='Business Rules' anchor='rules'>
+  <p>Typically a pair of XMPP servers connecting using this protocl will communcicate with multiple domains (e.g., a base domain and a MUC domain).   It is generally desirable to configure things so that all communications will share the same link, rather than establishing separate links for each domain.  Two or more connections MAY be initiated from one server to the other but this is NOT RECOMMENDED.   </p>
+  <p>
+    An XMPP server receiving messages over such a link should appropriately validate to and from elements of inbound messages.   The rules for this SHOULD be controlled by an priori agreement.  An inbound connection will generally be associated with several peer domains.   A RECOMMENDED approach is to consider each of these peers in turn and validate in the manner of a peer XMPP server connected using RFC 6020 for that domain.  In the event that an inbound message is not considered to be valid, it should be handled in a manner that this invalid message would be handled if it arrived over standard S2S.
+  </p>
+</section1>
+<section1 topic='Implementation Notes' anchor='impl'>
+  <section2 topic="General">
+    <p>
+      In simple terms, this can be considered as operation of RFC 6121 communication between a pair of XMPP servers without the preliminary negotiation done in RFC 6120.    It might be considered that the start point is the
+      DONE box in Figure 3 of RFC 6121.  The TCP connection is opened and messages start to flow.  All configuration informaition, including choice of port is handled by a the a priori configuration.</p>
```

Comment:
Ralph,

 

Thanks!   Fixed this and the other error you pointed out.   Also ran spell check (thought I’d done this) which found a couple of errors and I hope is now dealing with your spell check comment.

 

Change pushed to GitHub, and I now need to ask Kev about the next Git step!

 

 

Steve

 

> +      DONE box in Figure 3 of RFC 6121.  The TCP connection is opened and messages start to flow.  All configuration informaition, including choice of port is handled by a the a priori configuration.</p>

That figure is in RFC 6120

—
Reply to this email directly or view it on GitHub <https://github.com/xsf/xeps/pull/11/files#r33967286> .  <https://github.com/notifications/beacon/AMbljLOy8um8BaIwKnBXUpS-Z15-YKdxks5oasJcgaJpZM4FS25z.gif>